sys/dev/drm/drmkpi/include/linux/kthread.h

```diff
@@ -46,7 +46,7 @@ struct kthr_wrap {
 	struct thread *__td;						\
 	struct kthr_wrap *w;						\
 									\
-	w = malloc(sizeof(struct kthr_wrap), M_DRMKMALLOC, 0);		\
```

Comment:
Is it ok? Linux seems to think kthread_run can fail, so M_NOWAIT and checking for NULL here seems fine? Especially since Linux documents it as returning ERR_PTR(-ENOMEM).

Also, on error where you set __td to NULL below, it needs to be ERR_PTR(-EFOO), probably even `ERR_PTR(-what_kthread_add_returned)`. NULL does not count as an error for IS_ERR, only IS_ERR_OR_NULL.

Reply:
on my opinion the general-purpose approach is always WAITOK, unless there is strong reason to NOWAIT, like a non-sleepable lock etc.
E.g. if we have multiple malloc NOWAIT in the row during panfrost driver attachment and lack of resources, then we may never attach the driver.
In reality that does not matter as we attach driver once in boot time when plenty of resource available ?
may be linux have more usages of kthread_run, but we have only on panfrost attachment, when it creates scheduler thread.
what do you think?

Reply:
In general I prefer using `M_WAITOK` if it's safe to do so.  I agree with Jess on fixing the error return to return the error from kthread_add.

Reply:
Return code fixed. I have verified error path works.